pkg/api/helpers/upload.go

```diff
@@ -0,0 +1,108 @@
+// Package helpers provides useful wrappers for clients using the lakeFS OpenAPI.
+package helpers
+
+import (
+	"github.com/treeverse/lakefs/pkg/api"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// ClientUpload uploads contents as a file using client-side ("direct") access to underlying
+// storage.  It requires credentials both to lakeFS and to underlying storage, but
+// considerably reduces the load on the lakeFS server.
+func ClientUpload(ctx context.Context, client api.ClientWithResponsesInterface, repoID, branchID, filePath string, metadata map[string]string, contents io.ReadSeeker) (*api.ObjectStats, error) {
+	resp, err := client.GetPhysicalAddressWithResponse(ctx, repoID, branchID, &api.GetPhysicalAddressParams{
+		Path: filePath,
+	})
+	if err != nil {
```

Comment:
Not part of this PR, but is this the recommended way to check for errors from the client?
3 Different checks just to make sure nothing failed..

Reply:
No idea.  @nopcoder WDYT?